sorc/ncep_post.fd/MDL2P.f

```diff
@@ -1816,6 +1819,101 @@ SUBROUTINE MDL2P(iostatusD3D)
           ENDIF
         ENDIF
 !     
+!***  CHIPSI
+!
+        IF (IGET(1021) > 0 .or. IGET(1022) > 0) THEN
+          IF (LVLS(LP,IGET(1021)) > 0 .or. LVLS(LP,IGET(1022)) > 0) THEN
+            CALL CALCHIPSI(USL,VSL,CHI,PSI)
+!         print *,'me=',me,'EGRID1=',EGRID1(1:10,JSTA)
+!     
+!*** CHI 
+!
+          IF (LVLS(LP,IGET(1021)) > 0) THEN  
+!$omp  parallel do private(i,j)
+             DO J=JSTA,JEND
+               DO I=ISTA,IEND
+                 GRID1(I,J) = CHI(I,J)
+               ENDDO
+             ENDDO
+
+            IF (SMFLAG .or. ioform == 'binarympiio' ) THEN
+              call AllGETHERV(GRID1)
+              if (ioform == 'binarympiio') then
+!               nsmooth = max(2, min(30,nint(jm/94.0)))
+!             do k=1,5
+                CALL SMOOTHC(GRID1,SDUMMY,IM,JM,0.5)
+                CALL SMOOTHC(GRID1,SDUMMY,IM,JM,-0.5)
+!             enddo
+              else
+                NSMOOTH = nint(4.*(13500./dxm))
+!             endif
+              do k=1,NSMOOTH
+                CALL SMOOTH(GRID1,SDUMMY,IM,JM,0.5)
+              end do
+              endif
+            ENDIF
```

Comment:
@KarinaAsmar-NOAA To improve runtime, comment out these lines.

Reply:
@WenMeng-NOAA I commented out the lines and verified, the results without smoothing look the same, at least visually.